test/functional/wallet_disableprivkeys.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test running bitcoind with -disableprivatekeys options.
+
+- Start a wallet with -disableprivatekeys and make sure we can't execute private key operations
+- Test that disabeling the option later (or enabling after creation) fails.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class DisablePrivKeysTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [[], ["-disableprivatekeys"]]
+
+    def run_test(self):
+        assert_equal(self.nodes[1].getwalletinfo()['keypoolsize'], 0)
+        self.sync_all()
+        self.nodes[0].generate(101)
+        self.sync_all()
+
+        n0addrR = self.nodes[0].getnewaddress()
+        n0pubkR = self.nodes[0].getaddressinfo(n0addrR)['pubkey']
+        n0addrC = self.nodes[0].getnewaddress()
+        n0pubkC = self.nodes[0].getaddressinfo(n0addrC)['pubkey']
+        n0addr  = self.nodes[0].getnewaddress()
+        self.nodes[0].sendtoaddress(n0addrR, 10)
+        self.nodes[0].generate(6)
+        self.sync_all()
+
+        assert_raises_rpc_error(-4,"Error: Private keys are disabled (-disableprivatekeys)", self.nodes[1].getnewaddress)
+        assert_raises_rpc_error(-4,"Error: Private keys are disabled (-disableprivatekeys)", self.nodes[1].getrawchangeaddress)
+
+        self.nodes[1].importpubkey(n0pubkR) #TODO switch to importmulti
+        self.nodes[1].importpubkey(n0pubkC)
+        assert_equal(self.nodes[1].getbalance("*", 1, True), 10)
```

Comment:
Unsure why you're explicitly giving the `"*", 1, True` and not just doing `getbalance()` here.